cosmwasm/packages/wasmi-runtime/src/wasm/contract_operations.rs

```diff
@@ -53,12 +53,12 @@ pub fn init(
         EnclaveError::FailedToDeserialize
     })?;
 
-    let contract_key = generate_encryption_key(&parsed_env, contract)?;
+    let contract_address = &parsed_env.contract.address;
+    let contract_key = generate_encryption_key(&parsed_env, contract, contract_address.as_slice())?;
```

Comment:
The original point was to bind the contract address to the contract key, which was useful in the first iteration when we were checking the contract key, not the code hash. In this context it isn't forgeable, since offline attacks are not relevant to this use.

It isn't important anymore, but I left it there because it was already written, and it doesn't hurt like you said